diff --git a/start-stop-daemon/start-stop-daemon.8 b/start-stop-daemon/start-stop-daemon.8
index de2d35c3..de666e65 100644
--- a/start-stop-daemon/start-stop-daemon.8
+++ b/start-stop-daemon/start-stop-daemon.8
@@ -20,7 +20,7 @@
 .\" You should have received a copy of the GNU General Public License
 .\" along with this program.  If not, see <https://www.gnu.org/licenses/>.
 .
-.TH start\-stop\-daemon 8 "2017-07-04" "Debian Project" "dpkg suite"
+.TH start\-stop\-daemon 8 "2017-08-12" "CRUX 3.3" "core services"
 .nh
 .SH NAME
 start\-stop\-daemon \- start and stop system daemon programs
diff --git a/start-stop-daemon/start-stop-daemon.c b/start-stop-daemon/start-stop-daemon.c
index 81357504..f586fd91 100644
--- a/start-stop-daemon/start-stop-daemon.c
+++ b/start-stop-daemon/start-stop-daemon.c
@@ -20,10 +20,34 @@
  * Changes by Ian Jackson: added --retry (and associated rearrangements).
  */
 
+#if 0
 #include <config.h>
 #include <compat.h>
 
 #include <dpkg/macros.h>
+#else
+# define VERSION 		"20170812"
+# define CRUX 			"CRUX 3.3"
+
+# define HAVE_SYS_PARAM_H
+# define HAVE_SYS_SYSCALL_H
+# define HAVE_SYS_SYSCTL_H
+# define HAVE_SYS_USER_H
+# define HAVE_STDDEF_H
+# define HAVE_ERROR_H
+# define HAVE_ERR_H
+
+# define HAVE_CLOCK_MONOTONIC
+# define HAVE_GETDTABLESIZE
+# define HAVE_IOPRIO_SET
+# define HAVE_SETSID
+
+# define DPKG_ATTR_NORET	__attribute__((noreturn))
+# define DPKG_ATTR_PRINTF(X)
+
+# define _GNU_SOURCE
+# include <unistd.h>
+#endif
 
 #if defined(__linux__)
 #  define OS_Linux
@@ -159,6 +183,10 @@
 #define HAVE_IOPRIO_SET
 #endif
 
+#ifndef array_count
+# define array_count(x) 	(sizeof(x) / sizeof((x)[0]))
+#endif
+
 #define IOPRIO_CLASS_SHIFT 13
 #define IOPRIO_PRIO_VALUE(class, prio) (((class) << IOPRIO_CLASS_SHIFT) | (prio))
 #define IO_SCHED_PRIO_MIN 0
@@ -327,8 +355,7 @@ xstrndup(const char *str, size_t n)
 static void
 timespec_gettime(struct timespec *ts)
 {
-#if defined(_POSIX_TIMERS) && _POSIX_TIMERS > 0 && \
-    defined(_POSIX_MONOTONIC_CLOCK) && _POSIX_MONOTONIC_CLOCK > 0
+#ifdef HAVE_CLOCK_MONOTONIC
 	if (clock_gettime(CLOCK_MONOTONIC, ts) < 0)
 		fatal("clock_gettime failed");
 #else
@@ -646,9 +673,9 @@ usage(void)
 static void
 do_version(void)
 {
-	printf("start-stop-daemon %s for Debian\n\n", VERSION);
-
-	printf("Written by Marek Michalkiewicz, public domain.\n");
+	printf("start-stop-daemon " VERSION " for " CRUX "\n\n"
+		"Written by Marek Michalkiewicz, public domain.\n"
+		"Adjusted for " CRUX ".\n");
 }
 
 static void DPKG_ATTR_NORET
